packages/@eventual/core/src/socket.ts

```diff
@@ -0,0 +1,81 @@
+import type { FunctionRuntimeProps } from "./function-props.js";
+import { CallKind, createCall, type SocketSendCall } from "./internal/calls.js";
+import { registerEventualResource } from "./internal/resources.js";
+import { isSourceLocation, type SocketSpec } from "./internal/service-spec.js";
+import { parseArgs } from "./internal/util.js";
+import type { ServiceContext } from "./service.js";
+
+export interface SocketHandlerContext {
+  socket: { socketName: string };
+  service: ServiceContext;
+}
+
+export type SocketHandlers = {
+  $connect: (
+    request: {
+      connectionId: string;
+      query?: Record<string, string | undefined>;
+    },
+    context: SocketHandlerContext
+  ) => Promise<void> | void;
+  $disconnect: (
+    request: { connectionId: string },
+    context: SocketHandlerContext
+  ) => Promise<void> | void;
+  $default: (
+    request: {
+      connectionId: string;
+      body?: string;
+      headers?: Record<string, string | undefined>;
+    },
+    context: SocketHandlerContext
```

Comment:
Should we drop the `$` prefix?

Reply:
the benefit of keeping them is that we could support custom route names in the same object.

{
   $connect: () => {},
   $disconnect: () => {},
   $default: () => {},
   default: () => {} // custom route key named default
}